Unset moment.js locale in timezone service

dombesz opened the pull request:
https://community.openproject.org/wp/48834

dombesz committed:
Unset moment.js locale in timezone service

oliverguenther reviewed (changes requested):
@dombesz Can you elaborate on why this would fix the issue? Removing these lines would prevent localized output of dates in e.g, the work package cards

dombesz commented:
The error is related to the localization mechanism in moment.js. In some languages, such as Arabic, Bengali, Burmese, Cambodian, Hindi etc, the numbers are also translated from traditional arabic `2024-01-30` to the language's native characters such as `៣០/០១/២០២៤`, `٣٠/‏١/‏٢٠٢٤` etc.
I just removed the localization of the date formatter to see how many tests would be affected by that.

After some analysis, the solution I see has 2 parts:
- Respect the user's language format from the settings `admin/settings/date_format` in the datepicker.
- Translate the dates back to a machine readable date format before sending them to the server.

oliverguenther commented:
Both your suggested changes would require a big change in the frontend (not saying I'm rejecting either, but would lean towards the second - using the Intl. APIs to access dates instead of moment).

You mentioned we could try to disable the locale setting just for parsing the ISO8601 dates (or use another mechanism to parse these).

dombesz committed:
Set locale in init-locale.ts and update it to avoid post formatting dates

Avoiding date postformatting is important, because in some languages the digit mapping is different.
Different digit maps cause issues while being sent to the server, we should always use the traditional Arabic numbers,
regardless of the chosen language.

Switching up the order of the loaded moment locales, so that english remains the default instead of german.

dombesz committed:
Add specs for datepicker with Arabic locale

dombesz committed:
Use moment.locale() to update locales to make sure the locales are case insensitive.

dombesz commented on frontend/src/app/core/datetime/timezone.service.ts:
```diff
@@ -37,13 +37,7 @@ export class TimezoneService {
-    this.setupLocale();
-  }
-
-  public setupLocale():void {
```
ℹ️This is no more required, because we set the locale in the `init-locale.ts` below.

dombesz committed:
Fix button translation in date_edit_field.rb

dombesz committed:
Add more explanation to the comments.
